tests/unit/src/main/scala/tests/BuildServerInitializer.scala

```diff
@@ -0,0 +1,110 @@
+package tests
+
+import scala.concurrent.Future
+import scala.util.Properties
+
+import scala.meta.internal.builds.BuildTool
+import scala.meta.internal.metals.Debug
+import scala.meta.internal.metals.Messages.ImportBuild
+import scala.meta.internal.metals.MetalsEnrichments._
+import scala.meta.internal.metals.ServerCommands
+
+import ch.epfl.scala.bsp4j.BspConnectionDetails
+import com.google.gson.Gson
+import org.eclipse.lsp4j.MessageActionItem
+
+trait BuildServerInitializer {
+  def initialize(layout: String, expectError: Boolean = false): Future[Unit]
+}
+
+trait QuickBuildInitializer extends BuildServerInitializer {
+  this: BaseLspSuite =>
+  override def initialize(
+      layout: String,
+      expectError: Boolean = false
+  ): Future[Unit] = {
+    Debug.printEnclosing()
+    writeLayout(layout)
+    QuickBuild.bloopInstall(workspace)
+    for {
+      _ <- server.initialize()
+      _ <- server.initialized()
+    } yield {
+      if (!expectError) {
+        server.assertBuildServerConnection()
+      }
+    }
+  }
+}
+
+trait BloopImportInitializer extends BuildServerInitializer {
+  this: BaseLspSuite =>
+  override def initialize(
+      layout: String,
+      expectError: Boolean = false
+  ): Future[Unit] = {
+    writeLayout(layout)
+    for {
+      _ <- server.initialize()
+      // Import build using Bloop
```

Comment:
It's probably somewhat obvious for people that are familiar with the process, but for someone not maybe we should instead just add some Scaladocs to these initializers. Something like

```
/**
 * Set up your workspace using QuickBuild as your build tool. This will take your `metals.json` file and quickly
 * produce `.bloop/` files from it.
 * /
QuickBuildInitializer
```

```
/**
 * Set up your workspace by responding to a Import Build request which will
 * run Bloop Install via the build tool being used.
 * /
BloopImportInitializer
 ```

```
/**
 * Assumes sbt is being used as a build tool and also for your BSP server. This sets
 * up the necessary sbt files and also generates the corresponding bsp config file.
 * /
SbtServerInitializer
 ```
 
Something like that more or less so it's really clear what is happening and suppose to happen with these.

Reply:
Thanks!